pymeasure/adapters/visa.py

```diff
@@ -228,7 +228,18 @@ def flush_read_buffer(self):
         in the read buffer and no END-indicator was present, read from the device until
         encountering an END indicator (which causes loss of data).
         """
-        self.connection.flush(pyvisa.constants.BufferOperation.discard_read_buffer)
+        try:
+            self.connection.flush(pyvisa.constants.BufferOperation.discard_read_buffer)
+        except NotImplementedError:
+            # fake discarding the read buffer by reading all available messages.
```

Comment:
Please add the additional context (that we need this because `flush` is currently only implemented for `asrl`) and a link to the upstream issue in pyvisa-py, so that we can re-check later and remove the workaround when support has been added to more (all?) backends.